### PR TITLE
Fix String.prototype.match() function to correctly change `lastValue`.

### LINKED ISSUE #119: String.prototype.match() does not properly change `lastIndex`.
```
var global = /./g;
global.lastIndex = { valueOf: function() { assert(false); } };
assert("X", "x".replace(global, function(a) { return "X"; }));
assert(0 == global.lastIndex);
```
The above assertions fail, due to `valueOf: function() { assert(false); }` being called, since lastIndex is not properly modified.

### PR DESCRIPTION
**Pull request by DanielBallaSZTE**
Fixes #119

Signed-off-by: Daniel Balla <dballa@inf.u-szeged.hu>

### CODE REVIEW

**Comment by DanielBallaSZTE** [comment]
Note: 
I do not exactly know why the line I removed was there before, (probably checking if `lastIndex` can be converted to an integer?) but the reason the testcase failed is that single line calling `valueOf` on the `lastIndex` property. If the RegExp is global, the `lastIndex` should always be set to `0` first.

**Comment by ksh8281** [comment]
That was for old vender test.
I don't remember case number, but old vender test testing how many times would execute toInteger.
I think the test is disappear

File: test/regression-tests/issue-119.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// Copyright 2015 the V8 project authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Tests that lastIndex of a global RegExp is overwritten as per
+// ECMA-262 6.0 21.2.5.8 step 10.c.
+
+var global = /./g;
+global.lastIndex = { valueOf: function() { assert(false); } };
+assert("X", "x".replace(global, function(a) { return "X"; }));
```
Comment (clover2123) [inline test/regression-tests/issue-119.js]: I think the comma should be replaced with `==` to check if the two characters are identical.

Comment (DanielBallaSZTE) [reply test/regression-tests/issue-119.js]: Thank you, somehow I missed that

**Comment by DanielBallaSZTE** [comment]
@clover2123 Fixed it, thank you!

**Pull request merged**